.pipelines/cni/load-test-templates/restart-hns-template.yaml

```diff
@@ -0,0 +1,29 @@
+parameters:
+  clusterName: ""
+  cni: "cniv1"
+
+steps:
+  - task: AzureCLI@1
+    inputs:
+      azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+      scriptLocation: "inlineScript"
+      scriptType: "bash"
+      addSpnToEnvironment: true
+      inlineScript: |
+        make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}
+        kubectl get pod -owide -A
+
+        echo "Validate pod IP assignment before HNS restart"
+        make test-validate-state CNI_TYPE=${{ parameters.cni }} OS_TYPE=windows
+
+        podList=`kubectl get pods -n kube-system -l os=windows -owide | grep privileged | awk '{print $1}'`
+        echo Restart HNS service on each node
+        for pod in $podList; do
+          kubectl exec -i -n kube-system $pod -- powershell Restart-Service hns
+        done
+
+        echo "Validate pod IP assignment after HNS restart"
+        make test-validate-state CNI_TYPE=${{ parameters.cni }} OS_TYPE=windows
```

Comment:
thinking to check more after HNS is restarted. For example, we can check Get-Service HNS from windows?

Reply:
From my findings the only thing that shows that HNS has been restarted was the[ event log ](https://github.com/Azure/azure-container-networking/blob/4def1fbe1de352f088f5b5c4bcf4bd7a93f8ac0e/.pipelines/cni/load-test-templates/restart-hns-template.yaml#L22C11-L22C133). I agree though that we should have more through checks/tests for HNS. For our purposes though the tests that follow the restart do confirm that HNS is working as intended.